hexrd/ui/main_window.py

```diff
@@ -217,25 +222,97 @@ def add_materials_panel(self):
                                            self.materials_panel.ui,
                                            'Materials')
 
-    def on_action_open_config_triggered(self):
+    def on_action_open_config_yaml_triggered(self):
+
         selected_file, selected_filter = QFileDialog.getOpenFileName(
             self.ui, 'Load Configuration', HexrdConfig().working_dir,
             'YAML files (*.yml)')
 
         if selected_file:
-            HexrdConfig().working_dir = os.path.dirname(selected_file)
-            HexrdConfig().load_instrument_config(selected_file)
+            path = Path(selected_file)
+            HexrdConfig().working_dir = str(path.parent)
+
+            HexrdConfig().load_instrument_config(str(path))
             self.update_config_gui()
 
-    def on_action_save_config_triggered(self):
+    def on_action_open_config_dir_triggered(self):
+        dialog = QFileDialog(self.ui, 'Load Configuration',
+                             HexrdConfig().working_dir,
+                             'HEXRD directories (*)')
+        dialog.setFileMode(QFileDialog.Directory)
+
+        selected_files = []
+        if dialog.exec_():
+            selected_files = dialog.selectedFiles()
+
+        if len(selected_files) == 0:
+            return
+        else:
+            selected_file = selected_files[0]
+
+        path = Path(selected_file)
+
+        def _load():
+            path = Path(selected_file)
+            HexrdConfig().working_dir = str(path.parent)
+
+            # Opening a .hexrd directory, so point to config.yml
+            path = path / 'config.yml'
+
+            HexrdConfig().load_instrument_config(str(path))
+
+        # Check we have the config.yml
+        if not (path / 'config.yml').exists():
+            msg = 'Invalid HEXRD directory, config.yml missing.'
+            QMessageBox.critical(self.ui, 'HEXRD', msg)
+            return
+
+        # We do this in a worker thread so the UI can refresh.
+        worker = AsyncWorker(_load)
+        worker.signals.finished.connect(self.update_config_gui)
+
+        self.thread_pool.start(worker)
+
+    def on_action_save_config_yaml_triggered(self):
+
         selected_file, selected_filter = QFileDialog.getSaveFileName(
             self.ui, 'Save Configuration', HexrdConfig().working_dir,
             'YAML files (*.yml)')
 
         if selected_file:
-            HexrdConfig().working_dir = os.path.dirname(selected_file)
+            HexrdConfig().working_dir = str(Path(selected_file).parent)
             return HexrdConfig().save_instrument_config(selected_file)
 
+    def on_action_save_config_dir_triggered(self):
+        dialog = QFileDialog(self.ui, 'Save Configuration',
+                             HexrdConfig().working_dir,
+                             'HEXRD directories (*)')
+        dialog.setFileMode(QFileDialog.Directory)
+        dialog.setAcceptMode(QFileDialog.AcceptSave)
+        dialog.setOptions(QFileDialog.ReadOnly)
```

Comment:
Unfortunately, it looks like it is still creating the non-suffixed directory for me. :slightly_frowning_face: 

If we can't stop it from doing that, I guess we can just not worry about the suffix.

Reply:
it is? I will check again